### ISSUE TITLE
Is it really the device orientation that is used or the current layout region?

### ISSUE THREAD

**Issue by kortenkamp** [reuschj/RotatableStack#7]
Thank you for providing RotatableStack!

I try to use it for a small toy project, and it works only partly: I have several views that are combined in a VStack. The first view consists of two Texts. They should be next to each other (HStack) in landscape, and on top of each other (VStack) in portrait. When I use RotatbleStack, they are rendered like a HStack all the time – and well, they are rendered into a landscape like view. For other views that are not "landscape-like" the switching works, though.

To illustrate this, here is the example using one RotableStack on the red circle and the "Discovered Wheels" views:
<img width="636" alt="image" src="https://github.com/user-attachments/assets/c5a3d356-9d2b-4016-b142-e323c8fe4bb0">

```
import SwiftUI
import RotatableStack

struct ContentView: View {
    @ObservedObject var bleManager = BLEManager()
    @State private var scan = false
    @State private var adv = false
    
    var body: some View {
        VStack {
            VStack  {
                HStack {
                        Text("Bluetooth")
                            .font(.headline)
                        Spacer()
                        if bleManager.isSwitchedOn {
                            Text("ON")
                                .foregroundColor(.green)
                        }
                        else {
                            Text("OFF")
                                .foregroundColor(.red)
                        }
                    }
                    HStack {
                        Text("Battery")
                            .font(.headline)
                        Spacer()
                        Text(bleManager.currentBattery)
                            .font(.headline)
                    }
            }.padding()
            
            RotatableStack {
                ZStack {
                    
                    Circle()
                        .stroke(Color(light: .black, dark: .white), lineWidth: 20)
                        .padding(10)
                    
                    Circle()
                        .fill(Color("wheelcolor"))
                        .padding(10)
                    
                    Text(bleManager.currentSpeed)
                        .foregroundColor(.white)
                        .font(.custom("Futura Bold", fixedSize: 70))
                }
                
                VStack {
                    Text("Discovered Wheels")
                        .font(.largeTitle)
                        .frame(maxWidth: .infinity, alignment: .center)
                    List(bleManager.peripherals) { peripheral in
                        Button(peripheral.name, action: {
                            self.bleManager.connectWheel(peripheral)
                        })
                    }
                }
            }
        }
    }
}
```

Now I would like to render the two text lines (Bluetooth and Battery) in a `HStack` instead of a `VStack` in landscape mode. Changing the according `VStack` into `HStack` shows the desired (correct) result for the landscape version:

<img width="635" alt="image" src="https://github.com/user-attachments/assets/e8036db1-14b0-4cf1-85ab-d4be9f3c7588">

But as it should be a `VStack` in portrait, I change that to `RotatableStack` and get this:

<img width="634" alt="image" src="https://github.com/user-attachments/assets/e2506b99-8605-4251-9ba6-6970104cdf3d">

So… this is not what I expect – it is neither the `HStack` nor the `VStack` result. 

Maybe I am just messing things up somewhere, but to me it looks like `RotableStack` only works in certain cases…

**Comment by reuschj** [comment]
If you check the source code, all it is does is wrap your content in a `GeometryReader` and check the dimensions of the parent geometry. If the parent geometry is wider than it is tall, it will use an `HStack` vs. `VStack`... if taller than wide, it uses `VStack`. However, it's looking at the shape of the content, not the device orientation. So, the device can be in portrait but your _content_ could still be landscape. This is what's happening in your example:

<img width="187" alt="01" src="https://github.com/user-attachments/assets/95ea06ec-48bb-4753-88f7-1a5f0deb46af">

The content is not using all available vertical height and is landscape... so uses the `HStack`. If you adjust the shape of the content to fill up the available space, it will use the `VStack` as expected.

I agree there may be some gotchas with this implementation. If you have a suggestions to make this more reliable, I'm more than willing to consider. Feel free to check out the source code and make suggestions via a PR if you want to try. Or fork and try an alternative approach.

(Basically, I don't have the bandwidth right now to dig into this too much, but def open to any collaboration that improves this library!)

**Comment by kortenkamp** [comment]
Thank you, @reuschj . I was going through this several times with many other solutions I found on the web, and it seems that there is no proper solution for my original problem, which is weird – the automatic layout should be able to take into account whether the *top-level* container is portrait or landscape (which could be the device, but also a floating window on iPad).

So, when I finally found RotatableStack I was hoping that it does what I am trying to achieve. However, I will try to find a way to get the geometry of the top level container somehow, if I manage to do that I will let you know. It is just a toy project for me, so I don't have the bandwidth as well...

**Comment by reuschj** [comment]
Makes sense! I think these are actually slightly different use cases. I actually prefer using the geometry as for the use case I designed it for, but can def see the use case for always following device orientation. ¯\_(ツ)_/¯ 

Ideally, both options would be available. If you find a good solution for using device orientation vs. geometry, I'd be happy to add it as alternative or option.
